src/Compilers/CSharp/Portable/CSharpResources.resx

```diff
@@ -7917,4 +7914,13 @@ To remove the warning, you can use /reference instead (set the Embed Interop Typ
   <data name="ERR_ClassIsCombinedWithRefStruct" xml:space="preserve">
     <value>Cannot allow ref structs for a type parameter known from other constraints to be a class</value>
   </data>
+  <data name="ERR_NotRefStructConstraintNotSatisfied" xml:space="preserve">
+    <value>The type '{2}' may not be a ref struct or a type parameter allowing ref structs in order to use it as parameter '{1}' in the generic type or method '{0}'</value>
```

Comment:
>The type '{2}' may not be a ref struct or a type parameter allowing ref structs in order to use it as parameter '{1}' in the generic type or method '{0}' [](http://example.com/codeflow?start=11&length=153)

Perhaps: "The type ... **must not be** a ref struct ... in order to use it as **type** parameter ..."

Reply:
For "may not be a", I think I used the following message as a template:
```
  <data name="ERR_LambdaInIsAs" xml:space="preserve">
    <value>The first operand of an 'is' or 'as' operator may not be a lambda expression, anonymous method, or method group.</value>
  </data>
```

For "in order to use it as parameter", there is a "prior art" too:
```
  <data name="ERR_RefConstraintNotSatisfied" xml:space="preserve">
    <value>The type '{2}' must be a reference type in order to use it as parameter '{1}' in the generic type or method '{0}'</value>
  </data>
```

I can make the suggested change if you still think it is needed. Please let me know.